include/deal.II/dofs/dof_vector.h

```diff
@@ -0,0 +1,272 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2007 - 2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__dof_vector_h
+#define dealii__dof_vector_h
+
+#include <deal.II/base/smartpointer.h>
+#include <deal.II/lac/constraint_matrix.h>
+#include <deal.II/lac/vector.h>
+#include <deal.II/lac/vector_memory.h>
+
+DEAL_II_NAMESPACE_OPEN
+/**
+ * An object storing a vector describing a finite element function
+ * together with its DoFHandler and the ConstraintMatrix associated
+ * with the handler.
+ *
+ * This class encapsulates a data vector with the finite element space
+ * information associated with it, such that every function which is
+ * not purely algebraic can assogiate the right finite element
```

Comment:
Did you mean to say "...such that every *vector* which is not purely algebraic..."?

I think I have a hunch what "purely algebraic" is supposed to mean (presumably just an element of R^n, without the connotation that the elements correspond to the nodal vectors of a finite element space) but I think that would be worthwhile making clearer.